### PR TITLE
Adds unittests and fixes bug with OP_SUCCESS

### PR DESCRIPTION
**Pull request by EthanHeilman**

### CODE REVIEW

File: src/test/data/script_tests.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
 ["0x01 0x0b", "11 EQUAL", "P2SH,STRICTENC", "OK", "push 1 byte"],
 ["0x02 0x417a", "'Az' EQUAL", "P2SH,STRICTENC", "OK"],
 ["0x4b 0x417a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a7a",
- "'Azzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz' EQUAL", "P2SH,STRICTENC", "OK", "push 75 bytes"],
+    "'Azzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz' EQUAL", "P2SH,STRICTENC", "OK", "push 75 bytes"],
```
Comment (arminsabouri) [inline src/test/data/script_tests.json]: This is unfortunately going to throw lint off 😔

File: src/test/data/script_tests.json [hunk 2]
```diff
@@ -250,7 +250,7 @@
 ["1", "NOP", "P2SH,STRICTENC,DISCOURAGE_UPGRADABLE_NOPS", "OK", "Discourage NOPx flag allows OP_NOP"],
 
 ["0", "IF NOP10 ENDIF 1", "P2SH,STRICTENC,DISCOURAGE_UPGRADABLE_NOPS", "OK",
- "Discouraged NOPs are allowed if not executed"],
+    "Discouraged NOPs are allowed if not executed"],
```
Comment (arminsabouri) [inline src/test/data/script_tests.json]: Something here

**Review by arminsabouri — COMMENTED** [review]
Wow these tests are looking great 
I think some formatting changes in script tests got in

File: src/test/opcat_tests.cpp [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    auto controlblocks = spendData.scripts[{witVerifyScript, TAPROOT_LEAF_TAPSCRIPT}];
+    auto controlblock = *(controlblocks.begin());
+
+    for (auto witDatum : witData) {
+        scriptWitness.stack.push_back(witDatum);
+    }
+    scriptWitness.stack.push_back(witVerifyScript);
+    scriptWitness.stack.push_back(controlblock);
+
+    // unsigned int flags = 0;
```
Comment (arminsabouri) [inline src/test/opcat_tests.cpp]: Nit: could remove this

File: src/test/opcat_tests.cpp [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    unsigned int maxElementSize = 522;
+    unsigned int maxDupsToCheck = 10;
+
+    std::vector<std::vector<unsigned char>> witData;
+    witData.push_back({});
+
+    for (unsigned int elementSize = 1; elementSize <= maxElementSize; elementSize++) {
+        std::vector<unsigned char> witVerifyScript;
+        // increase the size of stack element by one byte
+        witData.at(0).push_back(0x1A);
```
Comment (arminsabouri) [inline src/test/opcat_tests.cpp:126]: Any reason for this byte? 
Could we have done anything?

File: src/test/opcat_tests.cpp [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    // or not thrown as appropriate. 
+    unsigned int maxElementSize = 522;
+    unsigned int maxDupsToCheck = 10;
+
+    std::vector<std::vector<unsigned char>> witData;
+    witData.push_back({});
+
+    for (unsigned int elementSize = 1; elementSize <= maxElementSize; elementSize++) {
+        std::vector<unsigned char> witVerifyScript;
+        // increase the size of stack element by one byte
```
Comment (arminsabouri) [inline src/test/opcat_tests.cpp:125]: Would this increase the stack element size by one or add a new stack element altogether?

File: src/test/opcat_tests.cpp (hunk 4 above)
Comment (EthanHeilman) [reply src/test/opcat_tests.cpp:126]: Could be anything. "1A" is easier to count bytes when repeated than say 0xaa.

File: src/test/opcat_tests.cpp (hunk 5 above)
Comment (EthanHeilman) [reply src/test/opcat_tests.cpp:125]: This increases the size of the stack element, it does not increase the number of stack elements.

File: src/test/opcat_tests.cpp (hunk 3 above)
Comment (EthanHeilman) [reply src/test/opcat_tests.cpp]: Fixed

**Pull request merged**